src/WebUI/Common/KubeCardWithTable.tsx

```diff
@@ -0,0 +1,224 @@
+import { ArrayItemProvider } from "azure-devops-ui/Utilities/Provider";
+import * as React from "react";
+import { IVssComponentProperties } from "../Types";
+import "./KubeCardWithTable.scss";
+import { IResourceStatusProps, ResourceStatus } from "./ResourceStatus";
+import { format } from "azure-devops-ui/Core/Util/String";
+
+export interface ITableComponentProperties<T> extends IVssComponentProperties {
+    className?: string;
+    headingText?: string | JSX.Element;
+    headingDescription?: string;
+    hideHeaders?: boolean;
+    hideLines?: boolean;
+    items: T[];
+    columns: ITableColumn<T>[];
+    onItemActivated?: (event: React.SyntheticEvent<HTMLElement>, tableRow: ITableRow<any>, selectedItem: any) => void;
+    onItemSelected?: (event: React.SyntheticEvent<HTMLElement>, tableRow: ITableRow<any>, selectedItem: any) => void;
+}
+
+export class KubeCardWithTable<T> extends BaseComponent<ITableComponentProperties<T>> {
+    public render(): React.ReactNode {
+        return (
+            <CustomCard className={css("flex-grow", "bolt-card-no-vertical-padding", this.props.className || "")}>
+                {
+                    this.props.headingText &&
+                    <CustomHeader>
+                        <HeaderTitleArea>
+                            <HeaderTitleRow>
+                                {
+                                    (typeof this.props.headingText === "string") ?
+                                        <HeaderTitle className="text-ellipsis" titleSize={TitleSize.Medium} >
+                                            {this.props.headingText}
+                                        </HeaderTitle> :
+                                        <HeaderTitle
+                                            className="text-ellipsis"
+                                            titleSize={TitleSize.Medium}
+                                            children={this.props.headingText}
+                                        />
+                                }
+                            </HeaderTitleRow>
+                            {
+                                this.props.headingDescription &&
+                                <HeaderDescription className="text-ellipsis">{this.props.headingDescription}</HeaderDescription>
+                            }
+                        </HeaderTitleArea>
+                    </CustomHeader>
+                }
+                <CardContent contentPadding={false}>
+                    {this.props.items && this.props.items.length > 0 && this._getComponent()}
+                </CardContent>
+            </CustomCard>
+        );
+    }
+
+    private _getComponent(): JSX.Element {
+        return (
+            <Table
+                className={"kube-list"}
+                itemProvider={new ArrayItemProvider<T>(this.props.items)}
+                columns={this.props.columns}
+                showHeader={!this.props.hideHeaders}
```

Comment:
is there a reason for not having the same prop name ? i.e. showHeader & showLines

Reply:
no other reason. this was a rename, not changing this.